src/metabase/models/interface.clj

```diff
@@ -704,6 +705,26 @@
   [_original-model dest-key _hydrated-key]
   [(u/->snake_case_en (keyword (str (name dest-key) "_id")))])
 
+(methodical/defmethod t2.hydrate/hydrate-with-strategy :around ::t2.hydrate/multimethod-simple
+  "Throws an error if do simple hydrations that make DB call on a sequence."
+  [model strategy k instances]
+  (if (or config/is-prod? (< (count instances) 2)
```

Comment:
```suggestion
  (if (or config/is-prod?
          (< (count instances) 2)
```

nit: weird to have two conditions on one line and the third on another, I'd do one line per condition